lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -786,15 +800,47 @@ listPools spl =
             apparentPerformance
 
 joinStakePool
-    :: forall n k.
+    :: forall ctx s t n k.
         ( PaymentAddress n k
+        , Buildable (ErrValidateSelection t)
+        , s ~ SeqState n k
         , k ~ ShelleyKey
+        , HardDerivation k
+        , ctx ~ ApiLayer s t k
         )
-    => ApiT PoolId
+    => ctx
+    -> StakePoolLayer IO
+    -> ApiT PoolId
     -> ApiT WalletId
     -> ApiWalletPassphrase
     -> Handler (ApiTransaction n)
-joinStakePool _ _ _ = throwError err501
+joinStakePool ctx spl (ApiT poolId) (ApiT wid) passwd = do
+    allPools <- listPools spl
+    case L.find (\(ApiStakePool pId _ _) -> pId == ApiT poolId) allPools of
+        Nothing ->
+            liftHandler $ throwE (ErrCreateCertNoSuchPool poolId)
+        Just _ -> do
+            let (ApiWalletPassphrase (ApiT pwd)) = passwd
+
+            selection <- liftHandler $ withWorkerCtx ctx wid liftE1 $ \wrk ->
+                W.createCert @_ @s @t wrk wid
+
+            (tx, meta, time, wit) <- liftHandler $ withWorkerCtx ctx wid liftE2 $ \wrk ->
+                W.signCert @_ @s @t @k wrk wid () pwd selection poolId
+
+            liftHandler $ withWorkerCtx ctx wid liftE3 $ \wrk ->
+                W.submitTx @_ @s @t @k wrk wid (tx, meta, wit)
```

Comment:
I need to think once again if submitTx can be used here, especially in the context of its interaction to DB